src/middlewared/middlewared/plugins/enclosure_/map.py

```diff
@@ -120,11 +140,15 @@ async def _map_enclosures(self, enclosures, slots):
             elements.append(element)
 
         info = await self.middleware.call("system.info")
+        if '-MINI-' in ["system_product"]:
+            model = info["system_product"]
+        else:
```

Comment:
Looking at this a bit closer, I'm curious why you're stripping `TRUENAS-` from the model name for all platforms except the mini?

Reply:
It's to deal with webui implementation.

Reply:
fixed

Reply:
Also previously some assumptions we would only need to map mini's. That's also why I removed comment on line 231 as I believe it's not accurate.

Reply:
Please resolve this if your happy

Reply:
Please make the following changes
```
-        info = await self.middleware.call("system.info")
+        mode = enclosures[0]["model"]
+        if not model:
+            model = await self.middleware.call("system.dmidecode_info")["system-product-name"]
+
         return [
             {
                 "id": "mapped_enclosure_0",
                 "name": "Drive Bays",
-                "model": info["system_product"],
+                "model": model,
                 "controller": True,
                 "elements": [
                     {

Reply:
Discussed with @darkfiberiru privately in slack. Changes were implemented here. Resolving this conversation.